enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,211 @@
+plugins via images, because this will enable offering plugins offline through an existing image registry.
+* Plugin owners must be able to easily distribute their binaries
+* Allow cluster-admins to control which plugins are offered to users
+  * This proposal is not concerned with _which_ plugins will be managed, as that is decided by cluster-admins
+* Controller for registering CRs with an API for listing, extracting and downloading plugins
+  * An API that generates an index that `krew` can consume via its custom index feature
+  * The index and binaries will served from a single route and service within the cluster
+
+### Non-Goals
+
+* The Controller will not build or package plugins
+* No recommending/limiting which plugins are served by the API
+* `krew` will not create or update the Plugin CRs, those will be managed by cluster-admins
+* No modifications to `oc` to support this functionality
+
+## Proposal
+
+Each component wishing to provide customers with their plugins will build and publish images via a trusted image registry
+and create a Plugin CR to provide an image name and the file path within that image for the binaries.
+Clients (i.e. `krew`) will read the index and download binaries from the Controller. The Controller is responsible for building the index from CRs, for pulling the images,
+extracting the binaries, and serving them to clients.
+
+* `krew` and `krew` plugins are upstream projects that Kubernetes users are already familiar with
+* A `krew`-compatible custom index can provide available plugins for a cluster in disconnected environments
+* The index will be served by a Controller with its contents managed by cluster-admins via CRs
+* The binaries will also be served by the Controller that will pull images from a trusted image registry and extract the binaries
+* With `krew index add https://someother-third-party-index` we won't limit users from adding their own index with whatever plugins they want
+* As `krew` itself is a `kubectl` plugin, it can be invoked using either using `kubectl krew` or `oc krew`
+
+Existing methods of downloading binaries (i.e. the console) will not be affected by this proposal. For the initial implementation, supported plugins will create Plugin CRs. The plugins will be downloaded and installed using `krew`.
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs and plugins available for Kubernetes/OpenShift and related services so that I can discover, install, and list them. If `odo` was made available by a cluster-admin, I could install it using:
+* `oc krew install odo`
+
+I could then interact with `odo`:
+* `oc odo --help`
+
+##### Example
+```text
+$ oc krew search odo
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+odo                  OpenShift Developer CLI     1.0          Not Installed
+```
+
+#### Story 2
```

Comment:
A good option I considered is that operators would inject their binaries information into that CR during installation. This would allow admins to have a pre-populated list that he can later modify.